File: compendium/ClientApp/src/app/models/spell.ts
import {HitEffect} from "./monster";

export class Spell {
    public id: string;
    public name: string;
    public school: SpellSchool;
    public castAsRitual: boolean;
    public level: number;
    public time: string;
    public range: string;
    public vocalComponent: boolean;
    public somaticComponent: boolean;
    public materials: string;
    public duration: string;
    public text: string;
    public classLists: string[];
    public isMultiTarget: boolean;
    public effects: HitEffect[];
    public castingTime: CastingTime;
    public atHigherLevels: string;
    public atHigherLevelEffects: HitEffect[];
}

export enum SpellSchool
{
    Abjuration = "Abjuration",
    Conjuration = "Conjuration",
    Divination = "Divination",
    Enchantment = "Enchantment",
    Evocation = "Evocation",
    Illusion = "Illusion =",
    Necromancy = "Necromancy",
    Transmutation = "Transmutation"
}

export enum CastingTime
{
    Action = "Action",
    BonusAction = "BonusAction",
    Reaction = "Reaction",
    AttackAction = "AttackAction",
    Minute = "Minute",
    _10Minutes = "_10Minutes",
    Hour = "Hour",
    _8Hours = "_8Hours",
    _12Hours = "_12Hours",
    _24Hours = "_24Hours",
    Varies = "Varies"
}

File: compendium/Models/CoreData/Spell.cs
using Compendium.Database;
using Compendium.Models.CoreData.Enums;

namespace Compendium.Models.CoreData
{
    public class Spell : ProjectKeyedDocument
    {
        public string Name { get; set; }
        public SpellSchool School { get; set; }
        public bool CastAsRitual { get; set; }
        public int Level { get; set; }
        public string Time { get; set; }
        public CastingTime? CastingTime { get; set; }
        public string Range { get; set; }
        public bool VocalComponent { get; set; }
        public bool SomaticComponent { get; set; }
        public string Materials { get; set; }
        public string Duration { get; set; }
        public string Text { get; set; }
        public List<string> ClassLists { get; set; } = new List<string>();
        public bool IsMultiTarget { get; set; }
        public List<HitEffect> Effects { get; set; }
        public string AtHigherLevels { get; set; }
        public List<HitEffect> AtHigherLevelEffects { get; set; }
    }

    public enum CastingTime
    {
        Action,
        BonusAction,
        Reaction,
        AttackAction,
        Minute,
        _10Minutes,
        Hour,
        _8Hours,
        _12Hours,
        _24Hours,
        Varies
    }
}
